### PR TITLE
[stable/superset] WIP: Refactoring, additional configs, secure secrets, versionbump

### PR DESCRIPTION
**Pull request by consideRatio**
I found various things I wanted to adjust while working with this helm chart, so I made this bigger PR for some initial discussion and local development. I can split it  up into reasonable chunks to get it merged later if it makes sense to you.

While reviewing, I recommend doing it one commit at the time.

## Change overview
- Some refactoring for readability, these changes are in alignment with common modern practices. See for example: https://github.com/helm/helm/pull/4562
- Added ability to add more configuration files
- Switched to using a k8s secret over the k8s configmap to save .Values.configFile content as sensitive passwords were suggested to be stored in the file.
- Added checksum annotations to restart pods on configuration changes
- Bumped superset version to 0.28.1

### CODE REVIEW

File: stable/superset/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 description: Apache Superset (incubating) is a modern, enterprise-ready business intelligence web application
 name: superset
-version: 1.0.0
-appVersion: "0.24.0"
+version: 1.0.1
+appVersion: "0.28.1"
```
Comment (cpanato) [inline stable/superset/Chart.yaml:4]: please update the readme.md file as well

Comment (cpanato) [reply stable/superset/Chart.yaml:4]: and bump the chart version to `1.1.0`

Comment (consideRatio) [reply stable/superset/Chart.yaml:4]: Done!

**Pull request merged**